### ISSUE TITLE
guess_bonds which guessing types

### ISSUE THREAD

**Issue by PicoCentauri** [MDAnalysis/mdanalysis#2143]
When you create a universe i.e with
```python
import MDAnalysis as mda
from MDAnalysis.tests.datafiles import PSF, DCD
u = mda.Universe(PSF,DCD, guess_bonds=True)
```
this will fail with an ValueError that the vdw radii for types are not defined. To resolve this one can first guess the types and then run bond guesser again. Maybe an option would be to add an option that the bond guesser can also try to guess the types and/or an option like: `u = mda.Universe(PSF,DCD, guess_bonds=True, guess_types=True)`.

**Comment by orbeckst** [comment]
Has this issue been addressed with the guessers PR #3704 ? @lilyminium @IAlibay @aya9aladdin  can you comment?

**Comment by lilyminium** [comment]
I believe so, although there is some suboptimal behaviour still (below). On an initial look, an error is still raised:

```
>>> u = mda.Universe(PSF, DCD, to_guess=("types", "bonds"))
/Users/lily/pydev/mdanalysis/package/MDAnalysis/coordinates/DCD.py:165: DeprecationWarning: DCDReader currently makes independent timesteps by copying self.ts while other readers update self.ts inplace. This behavior will be changed in 3.0 to be the same as other readers. Read more at https://github.com/MDAnalysis/mdanalysis/issues/3889 to learn if this change in behavior might affect you.
  warnings.warn("DCDReader currently makes independent timesteps"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/universe.py", line 465, in __init__
    self.guess_TopologyAttrs(
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/universe.py", line 1667, in guess_TopologyAttrs
    values = guesser.guess_attr(attr, fg)
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/guesser/base.py", line 163, in guess_attr
    return guesser_method()
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/guesser/default_guesser.py", line 438, in guess_bonds
    raise ValueError(
ValueError: vdw radii for types: 52, 23, 22, 24, 72, 30, 7, 1, 81, 25, 50, 32, 3, 54, 70, 2, 55, 21, 57, 6, 51, 31, 9, 29, 73, 56, 10, 20, 26, 5. These can be defined manually using the keyword 'vdwradii'
```

But I'm not sure this is an error to do with guessing itself -- it looks like guess_bonds has always relied on the `atoms.types` attribute to get vdw radii, and the PSFParser just reads the types from the file. If you forcibly re-guess the types from the names it works:

```
>>> u = mda.Universe(PSF, DCD, to_guess=("types", "bonds",), force_guess=("types",))
```

although the below doesn't, so possibly we should be prioritising "force_guess" attributes to go first (or enforcing that `force_guess` should be a subset of `to_guess`).

```
>>> u = mda.Universe(PSF, DCD, to_guess=("bonds",), force_guess=("types",))
/Users/lily/pydev/mdanalysis/package/MDAnalysis/coordinates/DCD.py:165: DeprecationWarning: DCDReader currently makes independent timesteps by copying self.ts while other readers update self.ts inplace. This behavior will be changed in 3.0 to be the same as other readers. Read more at https://github.com/MDAnalysis/mdanalysis/issues/3889 to learn if this change in behavior might affect you.
  warnings.warn("DCDReader currently makes independent timesteps"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/universe.py", line 465, in __init__
    self.guess_TopologyAttrs(
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/universe.py", line 1667, in guess_TopologyAttrs
    values = guesser.guess_attr(attr, fg)
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/guesser/base.py", line 163, in guess_attr
    return guesser_method()
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/guesser/default_guesser.py", line 438, in guess_bonds
    raise ValueError(
ValueError: vdw radii for types: 81, 57, 70, 54, 20, 72, 7, 52, 24, 21, 23, 6, 31, 1, 30, 50, 9, 32, 55, 2, 5, 3, 56, 29, 25, 10, 22, 73, 26, 51. These can be defined manually using the keyword 'vdwradii'
```

**Comment by lilyminium** [comment]
(Closed the issue since this Universe can now guess bonds on loading with the right combination and order of keywords.)

```
u = mda.Universe(PSF, DCD, to_guess=("types", "bonds",), force_guess=("types",))
```
